---
title: "Hemochromatosis blood samples NCBI"
author: "Janis Corona"
date: "1/1/2020"
output: html_document
---

These files are from https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE121620

Hemochromatosis is a disorder that can be fatal, it is when someone's
body has too much iron in it or their gut takes out more iron from food than normal it poison's the blood system.

This study analyzes 40-65 year old males and females who have two genotypes of 
homozygous strands and one of a heterozygous genotype for the hemochromatosis
gene compared to healthy 40-65 year old males and females. 


The platform for the GPL16686 family is 6.3 GB in file size
But a lot of the genes are NA, so read in the first 53,621 observations
the same as the length of the hemotochromatosis gene expressions file

```{r}
# hemo <- read.delim('GPL16686_family.soft', sep='\t', comment.char='#',
#                    na.strings=c('',' '), skip=8087, header=FALSE, nrows=53621)

iron <- read.csv('hemotochromatosisPatientsGeneExpressionValues.csv', sep=',',
                 header=TRUE, na.strings=c('',' '), skip=3)
ironMeta <- read.csv('hemotochromatosisPatientsGeneExpressionValues.csv', sep=',',
                     header=FALSE, na.strings=c('',' '), nrows=4)

# Hemo <- hemo[2:53621, c(1,6)]
# colnames(Hemo) <- c('ID','GB_ACC')
# Hemo <- Hemo[3:53619,]
# Hemo1 <- Hemo[complete.cases(Hemo$GB_ACC),]
# row.names(Hemo1) <- NULL

# combined <- merge(Hemo1, iron, by.x='ID', by.y='ID_REF')

```

The gene symbols can be taken from ensebl AFFY HuGene 2 0 st v1 probe in bioMart
this file is too large for Github and is 157.4 mb in size it is split apart below

```{r}
# BioMart <- read.delim('mart_export.txt', sep='\t', header=TRUE,
#                       na.strings=c('',' '))
# # # the following is the above file split into smaller file sizes to load in
# 
# mart1 <-bioMart[1:211142,]
# mart2 <-bioMart[211143:422284,]
# mart3 <-bioMart[422285:633426,]
# mart4 <-bioMart[633427:844569,]
# mart5 <-bioMart[844570:1055712,]
# mart6 <-bioMart[1055713:1266855,]
# mart7 <-bioMart[1266856:1477998,]
# mart8 <-bioMart[1477999:1689134,]
# 
# write.csv(mart1, 'mart1.csv', row.names=FALSE)
# write.csv(mart2, 'mart2.csv', row.names=FALSE)
# write.csv(mart3, 'mart3.csv', row.names=FALSE)
# write.csv(mart4, 'mart4.csv', row.names=FALSE)
# write.csv(mart5, 'mart5.csv', row.names=FALSE)
# write.csv(mart6, 'mart6.csv', row.names=FALSE)
# write.csv(mart7, 'mart7.csv', row.names=FALSE)
# write.csv(mart8, 'mart8.csv', row.names=FALSE)
###

mart1 <- read.csv('mart1.csv', sep=',', header=TRUE, na.strings=c('',' '))
mart2 <- read.csv('mart2.csv', sep=',', header=TRUE, na.strings=c('',' '))
mart3 <- read.csv('mart3.csv', sep=',', header=TRUE, na.strings=c('',' '))
mart4 <- read.csv('mart4.csv', sep=',', header=TRUE, na.strings=c('',' '))
mart5 <- read.csv('mart5.csv', sep=',', header=TRUE, na.strings=c('',' '))
mart6 <- read.csv('mart6.csv', sep=',', header=TRUE, na.strings=c('',' '))
mart7 <- read.csv('mart7.csv', sep=',', header=TRUE, na.strings=c('',' '))
mart8 <- read.csv('mart8.csv', sep=',', header=TRUE, na.strings=c('',' '))

BioMart <- rbind(mart1, mart2, mart3, mart4, mart5, mart6, mart7, mart8)

```

The biomart file has the probe IDs used in the platform but not by accession number. Merge the bioMart file with the iron file using the HGNC Symbol

```{r}
bio <- BioMart[,c(2:4)]
Bio <- bio[complete.cases(bio$HGNC.symbol),]
Bio <- Bio[!duplicated(Bio),]

Blend <- merge(iron, Bio, by.x='ID_REF', by.y='AFFY.HuGene.2.0.st.v1.probe')
# There are still duplicate HGNC names because of the probe ID identifying the gene
# where it is a pseudo gene

Blend1 <- Blend[,c(2:25,27)]

Blend_meta <- Blend[,26:27]

```

Group by genes and get the gene means so that there is only one unique gene in the table

```{r}
library(dplyr)
samples <- colnames(Blend1[1:24]) 
samples <- as.vector(samples)
geneMeans <- Blend1 %>% group_by(HGNC.symbol) %>% summarise_at(samples, mean, na.rm = TRUE)

```


```{r}
healthyMales <- as.data.frame(geneMeans[,c(1,20:22,25)])
healthyFemales <- as.data.frame(geneMeans[,c(1,23,24)])
```


Ignore the groups where 1 and 2 are one type of homozygous strand, 
and 3 is another or a heterozygous strand

```{r}
hemoMales <- as.data.frame(geneMeans[,c(1,2:7,9,10,12:15,19)])
hemoFemales <- as.data.frame(geneMeans[,c(1,8,11,16:18)])

```

Write these blood sample files out to csv to use later for comparisons with 
other blood samples for EGCG of overweight females and males with myocardial infarction.
```{r}
write.csv(geneMeans, 'hemochromatosis-cleaned.csv', row.names=FALSE)
write.csv(healthyMales, 'hemoHealthyMales.csv', row.names=FALSE)
write.csv(healthyFemales, 'hemoHealthyFemales.csv', row.names=FALSE)
write.csv(hemoMales, 'hemoMales.csv', row.names=FALSE)
write.csv(hemoFemales, 'hemoFemales.csv', row.names=FALSE)

```

Lets use dplyr to get the genes with the most and least fold change values between hemo males and healthy males, then hemo females and healthy females.But first separate the hemo males and females into their respective groups of 1,2, or 3 using the ironMeta table.
```{r}
#under 1000mikrongs/L blood serrum ferritin C282Y homozygous
group1_male <- as.data.frame(hemoMales[,c(1,9:12)]) 

#over 1000mikrogs/L blood serrum ferritin C282Y homozygous 
group2_male <- as.data.frame(hemoMales[,c(1,3,5,13,14)]) 

#H63D	homozygotes	and/or	H63D/C282Y	heterozygotes
group3_male <- as.data.frame(hemoMales[,c(1,2,4,6,7,8)]) 


group1_female <- as.data.frame(hemoFemales[,c(1,4,6)])
group2_female <- as.data.frame(hemoFemales[,c(1,3,5)])
group3_female <- as.data.frame(hemoFemales[,c(1,2)])


```

Get the gene means per group by taking the row means of each table, after removing the symbol field and making it the row names for the genes.
```{r}
row.names(group1_male) <- group1_male$HGNC.symbol
group1_male <- group1_male[,-1]

row.names(group2_male) <- group2_male$HGNC.symbol
group2_male <- group2_male[,-1]

row.names(group3_male) <- group3_male$HGNC.symbol
group3_male <- group3_male[,-1]

row.names(group1_female) <- group1_female$HGNC.symbol
group1_female <- group1_female[,-1]

row.names(group2_female) <- group2_female$HGNC.symbol
group2_female <- group2_female[,-1]


```

Turn the group3_female table from a matrix into a data frame, because when a df has only 1 field, R turns it into a vector of values instead of a df.
```{r}
row.names(group3_female) <- group3_female$HGNC.symbol
group3_female <- group3_female[,-1]

group3_female <- as.data.frame(group3_female)
row.names(group3_female) <- row.names(group2_female)
```

Do the same for the other tables of healthyFemales, healthyMales, hemoMales, and hemoFemales.
```{r}
row.names(healthyFemales) <- healthyFemales$HGNC.symbol
healthyFemales <- healthyFemales[,-1]

row.names(healthyMales) <- healthyMales$HGNC.symbol
healthyMales <- healthyMales[,-1]

row.names(hemoMales) <- hemoMales$HGNC.symbol
hemoMales <- hemoMales[,-1]

row.names(hemoFemales) <- hemoFemales$HGNC.symbol
hemoFemales <- hemoFemales[,-1]

```

Now use dplyr to find fold change values of the healthy males compared to all three groups of males.
```{r, warning=FALSE, error=FALSE, message=FALSE}
library(dplyr)
```


```{r}
group1_male$G1M_Mean <- rowMeans(group1_male)
group2_male$G2M_Mean <- rowMeans(group2_male)
group3_male$G3M_Mean <- rowMeans(group3_male)
group1_female$G1F_Mean <- rowMeans(group1_female)
group2_female$G2F_Mean <- rowMeans(group2_female)
group3_female$G3F_Mean <- rowMeans(group3_female)

healthyMales$HealthyMale_Mean <- rowMeans(healthyMales)
healthyFemales$HealthyFemale_Mean <- rowMeans(healthyFemales)
hemoMales$hemoMale_Mean <- rowMeans(hemoMales)
hemoFemales$hemoFemale_Mean <-rowMeans(hemoFemales)

```

Group all the table mean fields together into one table of its own.
```{r}
Means <- as.data.frame(cbind(group1_male$G1M_Mean, group2_male$G2M_Mean,
                             group3_male$G3M_Mean,
               group1_female$G1F_Mean, group2_female$G2F_Mean, group3_female$G3F_Mean,
               healthyFemales$HealthyFemale_Mean, healthyMales$HealthyMale_Mean,
               hemoFemales$hemoFemale_Mean, hemoMales$hemoMale_Mean))
row.names(Means) <- row.names(healthyFemales)
colnames(Means) <- c('G1M_Mean','G2M_Mean','G3M_Mean','G1F_Mean','G2F_Mean','G3F_Mean',
                     'healthyFemale_Mean',
                     'healthyMale_Mean','hemoFemale_Mean','hemoMale_Mean')

```

Add in the fold change values for each group with separate tables for each as the ratio of diseased group to healthy group.
```{r}
FC1 <- Means %>% mutate(FC1 = G1M_Mean/healthyMale_Mean)
FC2 <- FC1 %>% mutate(FC2 = G2M_Mean/healthyMale_Mean)
FC3 <- FC2 %>% mutate(FC3 = G3M_Mean/healthyMale_Mean)
FC4 <- FC3 %>% mutate(FC4 = G1F_Mean/healthyFemale_Mean)
FC5 <- FC4 %>% mutate(FC5 = G2F_Mean/healthyFemale_Mean)
FC6 <- FC5 %>% mutate(FC6 = G3F_Mean/healthyFemale_Mean)
FC7 <- FC6 %>% mutate(FC7 = hemoMale_Mean/healthyMale_Mean)
FC8 <- FC7 %>% mutate(FC8 = hemoFemale_Mean/healthyFemale_Mean)

row.names(FC8) <- row.names(Means)

All <- cbind(FC8, geneMeans[2:25])
```

```{r}
colnames(All)
```

Change the fold change fields to interpretable names.
```{r}
colnames(All)[11:18] <- c('FC_1m','FC_2m', 'FC_3m', 'FC_1F', 'FC_2F', 'FC_3F', 
                   'FC_malesOverall', 'FC_femalesOverall')
colnames(All)
```


Write this table of the fold change values and all samples and groups to csv.
```{r}
write.csv(All, 'all-fc-grops.csv', row.names=TRUE)
```

Now, lets find the most expressed genes overall for males
```{r}
most_males <- All[order(All$FC_malesOverall, decreasing=TRUE)[1:5],]
row.names(most_males)
```

Find the least expressed genes overall for males.
```{r}
least_males <- All[order(All$FC_malesOverall, decreasing=FALSE)[1:5],]
row.names(least_males)
```

Find the most expressed genes overall for females.
```{r}
most_females <- All[order(All$FC_femalesOverall, decreasing=TRUE)[1:5],]
row.names(most_females)
```

Find the least expressed genes overall for females.
```{r}
least_females <- All[order(All$FC_femalesOverall, decreasing=FALSE)[1:5],]
row.names(least_females)
```


What are the 5 most expressed genes in group 1 for males?
```{r}
g1m_most <- All[order(All$FC_1m, decreasing=TRUE)[1:5],]
row.names(g1m_most)
```

What are the 5 least expressed genes in group 1 for males?
```{r}
g1m_least <- All[order(All$FC_1m, decreasing=FALSE)[1:5],]
row.names(g1m_least)

```

What are the 5 most expressed genes in group 2 for males?
```{r}
g2m_most <- All[order(All$FC_2m, decreasing=TRUE)[1:5],]
row.names(g2m_most)

```

What are the 5 least expressed genes in group 2 for males?
```{r}
g2m_least <- All[order(All$FC_2m, decreasing=FALSE)[1:5],]
row.names(g2m_least)

```

What are the 5 most expressed genes in group 3 for males?
```{r}
g3m_most <- All[order(All$FC_3m, decreasing=TRUE)[1:5],]
row.names(g3m_most)

```

What are the 5 least expressed genes in group 3 for males?
```{r}
g3m_least <- All[order(All$FC_3m, decreasing=FALSE)[1:5],]
row.names(g3m_least)

```

What are the 5 most expressed genes in group 1 for females?
```{r}
g1f_most <- All[order(All$FC_1F, decreasing=TRUE)[1:5],]
row.names(g1f_most)

```

What are the 5 least expressed genes in group 1 for females?
```{r}
g1f_least <- All[order(All$FC_1F, decreasing=FALSE)[1:5],]
row.names(g1f_least)

```

What are the 5 most expressed genes in group 2 for females?
```{r}
g2f_most <- All[order(All$FC_2F, decreasing=TRUE)[1:5],]
row.names(g2f_most)


```

What are the 5 least expressed genes in group 2 for females?
```{r}
g2f_least <- All[order(All$FC_2F, decreasing=FALSE)[1:5],]
row.names(g2f_least)

```

What are the 5 most expressed genes in group 3 for females?
```{r}
g3f_most <- All[order(All$FC_3F, decreasing=TRUE)[1:5],]
row.names(g3f_most)

```

What are the 5 least expressed genes in group 3 for females?
```{r}
g3f_least <- All[order(All$FC_3F, decreasing=FALSE)[1:5],]
row.names(g3f_least)

```

When pulling the genes with the most fold change in MI males to healthy males, these were those five genes from a previous analysis: "THBS1" "VNN3"  "MERTK" "GPR15" "IFIT1". Are these genes in this data set?
```{r}
THBS1 <- All[grep('^THBS1$', row.names(All)),]
VNN3 <- All[grep('^VNN3$', row.names(All)),]
MERTK <- All[grep('^MERTK$', row.names(All)),]
GPR15 <- All[grep('^GPR15$', row.names(All)),]
IFIT1 <- All[grep('^IFIT1$', row.names(All)),]

most_MI_hemo_compared <- rbind(THBS1,VNN3,MERTK,GPR15,IFIT1)
most_MI_hemo_compared <- most_MI_hemo_compared[,c(11:18)]
most_MI_hemo_compared
```
Looking at the above data, all these same genes are for the most part overexpressed in hemochromatosis patients, except for group 2 males for THBS1 and the gene GPR15, which is under expressed in all groups except group 2 females.


```{r}
library(tidyr) #install.packages('tidyverse')

males <- most_MI_hemo_compared[,c(1:3,7)]

most_5 <- males %>% mutate(gene = row.names(males))
most_5 <- gather(most_5, 'group','foldChange',1:4)
most_5$foldChange <- round(most_5$foldChange,3)
head(most_5)
```

```{r}
library(ggplot2)

ggplot(data = most_5, aes(x=gene, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 1.4, by=.1), limits=c(0,1.4))+
  scale_fill_brewer(palette='OrRd') +
  ylab('Most Fold Change Genes in MI Male Samples')

```



The above bar plot shows the genes most expressed in males with myocardial infarction compared to healthy males, but these gene expression values in the plot are of hemochromatosis males compared to healthy males. Only the GPR15 gene is underexpressed in hemochromatosis males compared to healthy males and group 2 of the THBS1 gene.



When pulling the genes with the least fold change in MI males to healthy males, these were those five genes: "OR7D2" "KLRC3" "KLRC1" "IFNG"  "KLRC2"
```{r}
OR7D2 <- All[grep('^OR7D2$', row.names(All)),]
KLRC3 <- All[grep('^KLRC3$', row.names(All)),]
KLRC1 <- All[grep('^KLRC1$', row.names(All)),]
IFNG <- All[grep('^IFNG$', row.names(All)),]
KLRC2 <- All[grep('^KLRC2$', row.names(All)),]

least_MI_hemo_compared <- rbind(OR7D2,KLRC3, KLRC1,IFNG,KLRC2)
least_MI_hemo_compared <- least_MI_hemo_compared[,c(11:18)]
least_MI_hemo_compared

```




The above genes were all underexpressed in males with myocardial infarction compared to healthy males, but in hemochromatosis males only OR7D2 and KLRC3 are under expressed and IFNG for group 1 males only. The genes KLRC1 and KLRC2 are both over expressed in hemochromatosis males compared to healthy males.




```{r}

males <- least_MI_hemo_compared[,c(1:3,7)]

least_5 <- males %>% mutate(gene = row.names(males))
least_5 <- gather(least_5, 'group','foldChange',1:4)
least_5$foldChange <- round(least_5$foldChange,3)
head(least_5)
```

```{r}

ggplot(data = least_5, aes(x=gene, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 1.4, by=.1), limits=c(0,1.4))+
  scale_fill_brewer(palette='Paired') + 
  ylab('Least Fold Change in MI Male Samples')

```



The genes above are of the least expressed genes when comparing myocardial infarction males to healthy males, but with the gene expression values of hemochromatosis males compared to healthy males. Only the KLRC3 and OR7D2 genes are underexpressed the same as in MI males and also group 1 for the IFNG gene. Overall IFNG, KLRC1, and KLRC2 are actually overexpressed in hemochromatosis males compared to healthy males.



#### Compare the EGCG-green tea extract study of genes on overweight females.

What about the overweight females and the use of EGCG? Lets not exclude that group to compare these blood samples of hemochromatosis females to the overweight females most expressed genes and least expressed genes. The most expressed genes in the overweight females using EGCG were: "HBZ"       "FAU"       "RNA5SP449" "TMEM176A"  "TMEM176B"
```{r}
HBZ <- All[grep('^HBZ$', row.names(All)),]
FAU <- All[grep('^FAU$', row.names(All)),]
RNA5SP449 <- All[grep('^RNA5SP449$', row.names(All)),]
TMEM176A <- All[grep('^TMEM176A$', row.names(All)),]
TMEM176B <- All[grep('^TMEM176B$', row.names(All)),]

most_EGCG_hemo_compared <- rbind(HBZ,FAU,RNA5SP449,TMEM176A,TMEM176B)
most_EGCG_hemo_compared <- most_EGCG_hemo_compared[,c(11:18)]
most_EGCG_hemo_compared

```

```{r}

females <- most_EGCG_hemo_compared[,c(4:6,8)]

most_5 <- females %>% mutate(gene = row.names(females))
most_5 <- gather(most_5, 'group','foldChange',1:4)
most_5$foldChange <- round(most_5$foldChange,3)
head(most_5)

```

```{r}

ggplot(data = most_5, aes(x=gene, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 2, by=.2), limits=c(0,2))+
  scale_fill_brewer(palette='OrRd') +
  ylab('Most Fold Change Genes in EGCG Female Samples')

```



These genes above are the genes that were found to be over expressed in overweight females after using EGCG compared to before using EGCG. The expression values are for hemochromatosis females compared to healthy females. The gene HBZ is clearly over expressed as it was in the EGCG studies on overweight females. Since, HBZ is increased after using EGCG and it is shown here as having an over expression in females who produce toxic levels of iron, this could be bad for dieting if it also causes overweight females to produce toxic level of iron from their diets. The other genes stayed close to 1 for not much change, except for RNA5SP449 which was found under expressed in all females with hemochromatosis compared to healthy females.




The least expressed genes in fold change of overweight females compared to after using EGCG were: "VTRNA1-1"  "TMTC1"     "RNU4-2"    "LINC00189" "KIF4A"
```{r}
VTRNA1_1 <- All[grep('^VTRNA1-1$', row.names(All)),]
TMTC1 <- All[grep('^TMTC1$', row.names(All)),]
RNU4_2 <- All[grep('^RNU4-2$', row.names(All)),]
LINC00189 <- All[grep('^LINC00189$', row.names(All)),]
KIF4A <- All[grep('^KIF4A$', row.names(All)),]

least_EGCG_hemo_compared <- rbind(VTRNA1_1,TMTC1,RNU4_2,LINC00189,KIF4A)
least_EGCG_hemo_compared <- least_EGCG_hemo_compared[,c(11:18)]
least_EGCG_hemo_compared


```

One of the genes wasn't available in this hemochromatosis data set to compare, and that gene was the RNU4-2 gene.


```{r}

females <- least_EGCG_hemo_compared[,c(4:6,8)]

least_5 <- females %>% mutate(gene = row.names(females))
least_5 <- gather(least_5, 'group','foldChange',1:4)
least_5$foldChange <- round(least_5$foldChange,3)
head(least_5)

```

```{r}

ggplot(data = least_5, aes(x=gene, y=foldChange, fill=group)) +
  geom_bar(stat='identity', position=position_dodge())+
  scale_y_continuous(breaks = seq(0, 2, by=.2), limits=c(0,2))+
  scale_fill_brewer(palette='paired') + theme_minimal()+
  ylab('Least Fold Change Genes in EGCG Female Samples')

```


From the above plot the genes that were least expressed in the EGCG study of overweight females after using EGCG compared to before using EGCG are shown, but with the gene expression values of hemochromatosis females compared to healthy females. Group 1 of the LINC00189 gene and gene VTRNA1-1 are over expressed in hemochromatosis females compared to healthy females. EGCG made these genes under expressed when included in the diets of overweight females over 70 days. This could mean that VTRNA1-1 which is clearly over expressed in hemochromatosis females is amplified by more iron in the body, it is silenced more by EGCG, and/or some other possibility. Genecards.org summarisez this gene as: Vaults are large cytoplasmic ribonucleoproteins of about 13 MD. They are composed of a major vault protein, MVP (MIM 605088), 2 minor vault proteins, TEP1 (MIM 601686) and PARP4 (MIM 607519), and a nontranslated RNA component.